x-pack/plugins/security_solution/public/common/components/links/index.tsx

```diff
@@ -514,3 +514,54 @@ export const WhoIsLink = React.memo<{ children?: React.ReactNode; domain: string
 );
 
 WhoIsLink.displayName = 'WhoIsLink';
+
+interface SecuritySolutionLinkProps {
+  deepLinkId: SecurityPageName;
+  path?: string;
+}
+
+interface SecuritySolutionInjectedLinkProps {
+  onClick?: MouseEventHandler;
+  href?: string;
+}
+
+/**
+ * HOC that wraps any Link component and makes it a Security solutions internal navigation Link.
+ *
+ * It injects `onClick` and 'href' into the Link component calculated based on the` deepLinkId` and `path` parameters.
+ */
+export const withSecuritySolutionLink = <T extends {} & SecuritySolutionInjectedLinkProps>(
+  WrappedComponent: React.FC<T>
+): React.FC<Omit<T & SecuritySolutionLinkProps, 'onClick' | 'href'>> => {
+  const SecuritySolutionLink = ({
+    deepLinkId,
+    path,
+    ...rest
+  }: Omit<T & SecuritySolutionLinkProps, 'onClick' | 'href'>) => {
```

Comment:
I think we could do some types changes so it will be easier:
```suggestion
export const withSecuritySolutionLink = <T extends SecuritySolutionInjectedLinkProps>(
  WrappedComponent: React.FC<T>
): React.FC<T & SecuritySolutionLinkProps> => {
  const SecuritySolutionLink: React.FC<T & SecuritySolutionLinkProps> = ({
    deepLinkId,
    path,
    ...rest
  }) => {
```